src/cardano.h

```diff
@@ -46,11 +46,13 @@ STATIC_ASSERT(LOVELACE_MAX_SUPPLY < LOVELACE_INVALID, "bad LOVELACE_INVALID");
 #define MAX_HUMAN_ADDRESS_SIZE 150
 #define MAX_HUMAN_REWARD_ACCOUNT_SIZE 65
 
-#define MAINNET_PROTOCOL_MAGIC 764824073
 #define MAINNET_NETWORK_ID 1
+#define MAINNET_PROTOCOL_MAGIC 764824073
 
-#define TESTNET_PROTOCOL_MAGIC 1097911063
 #define TESTNET_NETWORK_ID 0
+#define TESTNET_PROTOCOL_MAGIC_LEGACY 1097911063
+#define TESTNET_PROTOCOL_MAGIC_PREPROD 1
+#define TESTNET_PROTOCOL_MAGIC_PREVIEW 2
```

Comment:
Nit - linking https://book.world.dev.cardano.org/environments.html might be useful